apps/condo/domains/billing/schema/BillingReceipt.js

```diff
@@ -87,6 +102,59 @@ const BillingReceipt = new GQLListSchema('BillingReceipt', {
                 return addValidationError(`${DV_UNKNOWN_VERSION_ERROR}dv] Unknown \`dv\``)
             }
         },
+        beforeChange: async ({
+            operation,
+            existingItem,
+            resolvedData,
+            context,
+        }) => {
+            const { sender: { fingerprint } } = resolvedData
+
+            if ('receiver' in resolvedData) {
+                return
+            }
+
+            let contextId
+            let recipient
+            if (operation === 'create') {
+                recipient = get(resolvedData, 'recipient')
+                contextId = get(resolvedData, 'context')
+            }
+            if (operation === 'update') {
+                recipient = get(resolvedData, 'recipient')
+                if (!recipient) {
+                    recipient = get(existingItem, 'recipient')
+                }
+                contextId = get(resolvedData, 'context')
+                if (!contextId) {
+                    contextId = get(existingItem, 'context')
+                }
+            }
+
+            let receiverId
```

Comment:
getOne is called on every api call :((((
May be we should decrease this amount by skipping some cases?
For example, if receiver is already specified...

How about `if(!existingItem.receiver || resolvedData.recepient)`

Reply:
Done!